pkg/virt-controller/services/vm.go

```diff
@@ -63,11 +64,18 @@ type vmService struct {
 }
 
 func (v *vmService) StartVMPod(vm *corev1.VM) error {
-
 	precond.MustNotBeNil(vm)
 	precond.MustNotBeEmpty(vm.GetObjectMeta().GetName())
+	precond.MustNotBeEmpty(vm.GetObjectMeta().GetNamespace())
 	precond.MustNotBeEmpty(string(vm.GetObjectMeta().GetUID()))
 
+	err := registrydisk.Initialize(vm, v.KubeCli)
```

Comment:
@fabiand the iscsi secrets feature and registry disks are completely independent things.

virt-controller is not involved at all for someone who is using iscsi auth for their VM explicitly. For example, all someone has to do to use iscsi auth is create the k8s secret and define the VM with an Auth field that references the k8s secret.  Only virt-handler will be involved in that flow.

The registry disk feature consumes the iscsi auth feature by dynamically generating k8s secrets in virt-controller.  That's what is being done in this initialization step. We're initializing everything that needs to be filled in before the VM is scheduled related to registry disks. 

Because the registry disk feature involves the VM's virt-launcher pod (where registry disks are launched), it seems logical that virt-controller coordinates these parts.